examples/pghostname_zig/src/main.zig

```diff
@@ -0,0 +1,30 @@
+const std = @import("std");
+const pgzx = @import("pgzx");
+
+comptime {
+    pgzx.PG_MODULE_MAGIC();
+
+    pgzx.PG_FUNCTION_V1("pghostname_zig", pghostname_zig);
+}
+
+fn pghostname_zig() ![]const u8 {
+    var buffer: [std.posix.HOST_NAME_MAX]u8 = undefined;
+    const hostname = try std.posix.gethostname(&buffer);
```

Comment:
```suggestion
    const hostname = std.posix.gethostname(&buffer) catch |err| switch (err) {
        error.PermissionDenied => {
            std.debug.print("Error getting hostname: permission denied\n", .{});
            return "unknown"[0..];
        },
        error.Unexpected => {
            std.debug.print("Error getting hostname: unexpected error\n", .{});
            return "unknown"[0..];
        },
    };
```

This compiles but returns empty hostname in SQL (the elog before return works fine). I suspect that's a bug in pgzx? 🤔

Reply:
Not sure why you get an empty hostname. I don't think this is pgzx, but Zig returning an empty string here. I haven't had any problems. But why do you need to return `"unknown"[0..]`? You are returning a string anyways.

For logging, don't use `std.debug.print`. Use `elog` to emit debug logs. This way you are using the postgres internal logging/reporting system.

Reply:
Fixed via `try pgzx.mem.PGCurrentContextAllocator.dupeZ(u8, hostname);`, thanks for the help in slack.